### ISSUE TITLE
UserStorageProvider Configuration not shown

### ISSUE THREAD

**Issue by dasniko** [keycloak/keycloak-ui#2989]
### Describe the bug

When using a custom UserStorageProvider implementation with the new admin-ui, the configuration isn't shown, whereas it is the case with the old/current admin-ui.

I tested it with the current nightly build and same provider code.
In "old" admin-ui, the config is properly displayed:
![user-storage-spi_old-admin-ui](https://user-images.githubusercontent.com/2075748/180200003-67e16059-c704-4e97-a7b4-c35c0ad920ce.png)

The new admin-ui doesn't show anything than the display name and the cache policy:
![user-storage-spi_new-admin-ui](https://user-images.githubusercontent.com/2075748/180200104-2c0c6634-7200-4387-a9b5-c6ff0691f8a6.png)

I wondered, because the built-in LDAP and Kerberos provider configs are properly displayed in the new ui...
Do I have to change something in the implementation of my storage provider?
Just in case you want to have a look into it, you can find the example [here](https://github.com/dasniko/keycloak-extensions-demo/tree/main/user-provider/src/main/java/dasniko/keycloak/user)

### Version

nightly (as of 2022-07-21)

### Expected behavior

all config properties defined in the factory are displayed properly

### Actual behavior

the config properties defined in the factory are missing

### How to Reproduce?

see description

### Anything else?

Didn't find a similar, already existing issue, but I don't want to believe this is not yet addressed...

**Comment by dasniko** [comment]
Same behavior with release 19.0.0

**Comment by bpedersen2** [comment]
From the code (https://github.com/keycloak/keycloak-admin-ui/blob/main/src/user-federation/ldap/LdapSettingsGeneral.tsx) it looks to me like this is all hand-coded.

**Comment by jonkoops** [comment]
@ssilvert is this something we want to prioritize?

**Comment by dasniko** [comment]
Thanks @bpedersen2 for looking into the code. I didn't do this because I blindly expected that the new admin ui has at least the same functionality as the old one. And as it works in a dynamic way there, it should also work the same way in the new ui.

Having this feature no more, makes the use and configuration of existing user storage providers impossible.

**Comment by jonkoops** [comment]
Considering this is missing functionality I am marking this as a critical issue that should preferably be included in the next patch release.

**Comment by jonkoops** [comment]
@agagancarczyk I am currently still quite busy with release process enhancements, could I ask you to take this one?

**Comment by djairhogeuens** [comment]
@ssilvert I'm a bit confused as I saw you changed the milestone for this issue to 20.0.0. Does this mean that it won't be included in a next patch release?

**Comment by jonkoops** [comment]
Most likely we will include all the work on `main` in a patch release as we landed a lot of important fixes. The milestone is only a placeholder for what will be prioritized next.

**Comment by ikcalB** [comment]
@edewit it looks like a fix has been merged - but not yet published as a patch release? 

we are on 19.0.1 and it seems this bug is still present. if so, when do you plan to release the next patch set?

**Comment by edewit** [comment]
Soon

**Comment by dasniko** [comment]
I'm afraid this is not yet finally fixed with `19.0.2` and should be re-opened (or should I create a new issue for it?).

While the dynamic fields are now properly rendered and filled with data, it's not possible to edit and submit data to the server. There is an exception thrown:
```
2022-09-14 10:57:28,941 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-5) Uncaught server error: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('https://kze2nq1l79.execute-api.eu-central-1.amazonaws.com/peanuts')
kcd_kc    |  at [Source: (io.quarkus.vertx.http.runtime.VertxInputStream); line: 1, column: 74] (through reference chain: org.keycloak.representations.idm.ComponentRepresentation["config"]->org.keycloak.common.util.MultivaluedHashMap["peanutsBaseUrl"])
kcd_kc    | 	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
kcd_kc    | 	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1728)
kcd_kc    | 	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1353)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromString(StdDeserializer.java:311)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.handleNonArray(StringCollectionDeserializer.java:284)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:192)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:182)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:25)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:609)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:437)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:32)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:176)
kcd_kc    | 	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
kcd_kc    | 	at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:2025)
kcd_kc    | 	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1175)
...
```

When comparing the payload which is sent from the UI to the server between old and new UI, I get the following:

**Old UI:**
```
{
    "name": "peanuts",
    "providerId": "peanuts-user-provider",
    "providerType": "org.keycloak.storage.UserStorageProvider",
    "parentId": "demo",
    "config": {
        "enabled": [
            "true"
        ],
        "cachePolicy": [
            "DEFAULT"
        ],
        "peanutsBaseUrl": [
            "https://someid.execute-api.eu-central-1.amazonaws.com/peanuts"
        ],
        "peanutsAuthUsername": [
            "peanuts"
        ],
        "peanutsAuthPassword": [
            "s3cr3t"
        ]
    }
}
```

**New UI:**
```
{
    "name": "peanuts",
    "config": {
        "peanutsBaseUrl": "https://someid.execute-api.eu-central-1.amazonaws.com/peanuts",
        "peanutsAuthUsername": "peanuts",
        "peanutsAuthPassword": "s3cr3t",
        "enabled": [
            "true"
        ],
        "cachePolicy": [
            "DEFAULT"
        ]
    },
    "providerId": "peanuts-user-provider",
    "providerType": "org.keycloak.storage.UserStorageProvider",
    "parentId": "demo"
}
```

So, the custom properties (in the example starting with "peanuts...", must be submitted as a string array, not a pure string.
Please check this also for all other types, not only for strings.

Thanks.

**Comment by edewit** [comment]
duplicate: #3341
